marshmallow_dataclass/__init__.py

```diff
@@ -367,6 +358,8 @@ def _field_by_type(
 ) -> Optional[Type[marshmallow.fields.Field]]:
     return (
         base_schema and base_schema.TYPE_MAPPING.get(typ)
+    ) or (
+        any(kw in str(typ) for kw in ["Serializable", "Mapping", "Dict"]) and dict
```

Comment:
kind of a hack but works pretty well. the issue was basically that we had a lot of types inheriting from TypedDict that marshmallow_dataclass didn't recognize as dicts